include/mysql/service_mysql_keyring.h

```diff
@@ -16,10 +16,72 @@
 #ifndef MYSQL_SERVICE_MYSQL_PLUGIN_KEYRING_INCLUDED
 #define MYSQL_SERVICE_MYSQL_PLUGIN_KEYRING_INCLUDED
 
+#ifndef MYSQL_ABI_CHECK
+#include <stdlib.h>
+
+// System keys cannot have ':' in their name. We use ':' as a separator between
+// system key's name and system key's verision
+// Keep adding keys' names to valid_percona_system_keys in sorted order. We later do binary_search
+// on this table. Also update valid_percona_system_keys_size.
+MY_ATTRIBUTE((unused)) static const size_t valid_percona_system_keys_size = 1;
```

Comment:
Considering the library-server separation, maybe the header file is not the best place for both code and data definitions. If server adds more keys, the older libraries will not work correctly, with no good reason. This would also help with MYSQL_ABI_CHECK above?

Reply:
I have moved it to mysys. Mysql sql/ seems like a better place for this function, but I think innodb will later also need access to those definitions.